modules/react/collection/lib/useBaseListModel.tsx

```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import {useVirtual} from './react-virtual';
 
-import {useUniqueId, assert, createModelHook, Generic} from '@workday/canvas-kit-react/common';
+import {useUniqueId, createModelHook, Generic} from '@workday/canvas-kit-react/common';
 
 export type Orientation = 'horizontal' | 'vertical';
 
 export const defaultGetId = (item: any): string => {
-  assert(item.id, 'A list item must have an `id` field or a `getId` function defined');
-  return item.id;
+  // assert(item.id, 'A list item must have an `id` field or a `getId` function defined');
```

Comment:
```suggestion
  assert(item.id, 'A list item must have an `id` field or a `getId` function defined');
```

Reply:
It should be removed. We cannot make guarantees on unloaded data.

Reply:
ok that was my initial suggestion, just wasn't sure